### DISCUSSION TITLE
Error running a simulation

### DISCUSSION THREAD

**Discussion by yvalter** [simnibs/simnibs#356] [8. Simulation workflows] [answered]
I received the following that prevented a TMS simulation:

CRITICAL: Uncaught exception
Traceback (most recent call last):
  File "C:\ ... \Desktop\SimNibs\simnibs_env\lib\site-packages\simnibs\utils\cond_utils.py", line 164, in test_numerical
    c = float(v)
TypeError: float() argument must be a string or a number, not 'NoneType'

During handling of the above exception, another exception occurred:

TypeError: The value -1 in cond_list is not numerical

**Comment by TorgeW** [comment]
Could you please post the script you are using to run the simulation (if you are not using the GUI) as well as the full simnibs_simulation_....log and simnibs_simulation_....mat files

**Comment by yvalter** [comment]
I should add that the head was segmented using TAP. 

Below are the simulation and log files as well as a screenshot of the GUI used for simulation 

![Screenshot_gui_2](https://github.com/simnibs/simnibs/assets/121051996/22884e16-aba3-4268-884a-da4046528010)
[simnibs_simulation_20240607-093603.log](https://github.com/user-attachments/files/15740548/simnibs_simulation_20240607-093603.log)
[simnibs_simulation_20240607-093603.zip](https://github.com/user-attachments/files/15740586/simnibs_simulation_20240607-093603.zip)

**Reply by TorgeW** [reply, marked as answer]
It looks like there are elements with the tag 0 in your mesh. That is not a valid tag. (The eeg cap positions are also not supposed to be so far off the head but sit right on top)

You can confirm that by opening m2m_a.msh in gmsh and clicking "Tools" -> "Visibility" which will show you all tags present in your mesh.
